File: VB/MySolution.Module/BusinessObjects/Attributes.vb
Imports DevExpress.ExpressApp.DC
Imports DevExpress.Persistent.Base
Imports System
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

<MetadataType(GetType(EmployeeMetadata)), _
DefaultClassOptions, ImageName("BO_Employee")> _
Partial Public Class Employee
End Class
Public Class EmployeeMetadata
    Public Property Id() As Int32
End Class

<MetadataType(GetType(TaskMetadata)), _
DefaultClassOptions, ImageName("BO_Task")> _
Partial Public Class Task
End Class
Public Class TaskMetadata
    Public Property Id() As Int32

    Public Property Description() As String
End Class

File: CS/MySolution.Module/BusinessObjects/Attributes.cs
using DevExpress.ExpressApp.DC;
using DevExpress.Persistent.Base;
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace MySolution.Module.BusinessObjects {

    [MetadataType(typeof(EmployeeMetadata))]
    [DefaultClassOptions, ImageName("BO_Employee")]
    public partial class Employee { }
    public class EmployeeMetadata {
        [Browsable(false)]
        public Int32 Id { get; set; }
    }

    [MetadataType(typeof(TaskMetadata))]
    [DefaultClassOptions, ImageName("BO_Task")]
    public partial class Task { }
    public class TaskMetadata {
        [Browsable(false)]
        public Int32 Id { get; set; }

        [FieldSize(FieldSizeAttribute.Unlimited)]
        public String Description { get; set; }
    }
}

